views/map.js

```diff
@@ -185,21 +175,33 @@ module.exports = View.extend({
 
   addPinToMap: function(xPos, yPos) {
     var newPinBox = domify(pinTemplate())
-    this.query('.draw-area').appendChild(newPinBox)
 
-    newPinBox.style.top =  yPos
-    newPinBox.style.left =  xPos
+    if (this.query('.draw-area').children.length === 0) {
+      newPinBox.classList.add('start-pin')
+    }else if (this.query('.draw-area').children.length === 1) {
+      newPinBox.classList.add('finish-pin')
+    }
+    if (this.query('.draw-area').children.length < 2) {
+      //Add new pin to draw-area
+      this.query('.draw-area').appendChild(newPinBox)
+
+      newPinBox.style.top =  yPos + 'px'
+      newPinBox.style.left =  xPos + 'px'
 
-    var invalue = [
-      'scale(' + 1 / this.scale + ', ' + 1 / this.scale + ')'
-    ]
-    applyTransform(newPinBox, invalue.join(' '))
+      var invalue = [
+        'scale(' + 1 / this.scale + ', ' + 1 / this.scale + ')'
+      ]
+      applyTransform(newPinBox, invalue.join(' '))
+    }else {
+      //If already 2 pins exist -> delete all
+      this.removeAllPins()
+    }
   },
 
   removeAllPins: function() {
     var pins = document.getElementsByClassName('pin');
-    for (var i = 0; i < pins.length; i++) {
-      pins[i].remove()
+    while (pins.length > 0) {
+      pins[0].remove()
     }
```

Comment:
Weil sich nach dem löschen aus dem pins array auch dessen länge ändert und damit auch jedes mal die bedingung für die for schleife anders ist → es wird nicht alles gelöscht bei for!